src/Form/Control/Calendar.php

```diff
@@ -4,78 +4,102 @@
      */
     public $options = [];
 
     /**
-     * Allow to set Calendar.js function.
+     * Set flatpickr option.
      */
     public function setOption($name, $value)
     {
         $this->options[$name] = $value;
     }
 
-    protected function renderView(): void
+    /**
+     * Load flatpickr locale file.
+     * Pass it has an option when adding Calendar input.
+     *  Form\Control\Calendar::requireLocale($app, 'fr');
+     *  $form->getControl('date')->options['locale'] = 'fr';.
+     */
+    public static function requireLocale(App $app, string $locale)
     {
-        if (!$this->icon) {
-            switch ($this->type) {
-            //case 'date': $this->icon = '
-            }
-        }
+        $app->requireJs($app->cdn['flatpickr'] . '/l10n/' . $locale . '.js');
+    }
 
-        if (!$this->type) {
-            $this->type = 'datetime';
-        }
+    /**
+     * Apply locale globally to all flatpickr instance.
+     */
+    public static function setLocale(App $app, string $locale)
+    {
+        self::requireLocale($app, $locale);
+        $app->html->js(true, (new JsChain('flatpickr'))->localize((new JsChain('flatpickr'))->l10ns->{$locale}));
+    }
 
-        if ($this->readonly) {
-            $this->options['onShow'] = new JsFunction([new JsExpression('return false')]);
-        }
+    /**
+     * Set first day of week for calendar display.
+     * Applied globally to all flatpickr instance.
+     */
+    public static function setDayOfWeek(App $app, int $day)
+    {
+        $app->html->js(true, (new JsExpression('flatpickr.l10ns.default.firstDayOfWeek = [day]', ['day' => $day])));
+    }
+
+    protected function init(): void
+    {
+        parent::init();
 
-        $typeFormat = $this->type . '_format';
-        if ($format = $this->app->ui_persistence->{$typeFormat}) {
-            $formatter = 'function(date, settings){
-                            if (!date) return;
-                            return atk.phpDate([format], date);
-                        }';
-            $this->options['formatter'][$this->type] = new JsExpression($formatter, ['format' => $format]);
+        // Get ui persitense default.
+        if ($options = $this->app->ui_persistence->calendar_options) {
+            array_merge($options, $this->options);
         }
 
-        $this->options['type'] = $this->type;
+        // get format from ui persistence if not set.
+        $this->options['dateFormat'] = $this->options['dateFormat'] ?? $this->app->ui_persistence->{$this->type . '_format'};
 
-        if ($dayOfWeek = $this->app->ui_persistence->firstDayOfWeek) {
-            $this->options['firstDayOfWeek'] = $dayOfWeek;
+        // set default according to type.
+        switch ($this->type) {
+            case 'time':
+                $this->options['enableTime'] = true;
+                $this->options['noCalendar'] = true;
+                $this->options['time_24hr'] = $this->use24HrTime;
```

Comment:
no property for 24h - parse php date format like:

```
private function is24hrTimeFormat(): bool
{
    return !preg_match('~[gh]~');
}
```

Reply:
this is not for date parsing but rather an option for flatpickr on how to display/set time for user.

Reply:
if there is a format "option for flatpickr on how to display/set time for user", then parse if 24hr format from it

Reply:
I think they are independent (setting vs output). But I will check and change accordingly. I will also need to add tests for this.

Reply:
option is now set according to php format and test is added.